file_scan.c

```diff
@@ -660,7 +660,7 @@ static void csum_whole_file_init(GMutex **mutex, void *location,
 	unsigned long long cur_scan_files;
 	*mutex = g_dataset_get_data(location, "mutex");
 
-	cur_scan_files = __sync_add_and_fetch(&_cur_scan_files, 1);
+	cur_scan_files = __atomic_fetch_add(&_cur_scan_files, 1, __ATOMIC_SEQ_CST);
```

Comment:
I've been looking at the Debian package for duperemove and I noticed that this patch actually changes the behavior of the program slightly because these atomics return different values.

* `__sync_add_and_fetch` returns the value *after* being incremented.
* `__atomic_fetch_add` returns the value *before* being incremented.

Was this an oversight? I don't think the damage is too bad because these are only used inside `printfs`.

If this needs changing, thankfully there is a `__atomic_add_fetch` function which will use the old behavior.